lib/pure/nativesockets.nim

```diff
@@ -438,17 +438,13 @@ proc getHostname*(): string {.tags: [ReadIOEffect].} =
   ## Returns the local hostname (not the FQDN)
   # https://tools.ietf.org/html/rfc1035#section-2.3.1
   # https://tools.ietf.org/html/rfc2181#section-11
-  const size = 64
-  result = newString(size)
-  when useWinVersion:
-    let success = winlean.gethostname(result, size)
-  else:
-    # Posix
-    let success = posix.gethostname(result, size)
-  if success != 0.cint:
-    raiseOSError(osLastError())
-  let x = len(cstring(result))
-  result.setLen(x)
+  var buf {.noinit.}: array[256, char]
+  if gethostname(buf[0].addr, buf.len.cint) != 0: raiseOSError(osLastError())
+  addCstring(result, buf)
+  if result.len == buf.len:
+    raise newException(OSError, "getHostname() returned name too long")
```

Comment:
Why not reallocate instead and try again?

Reply:
According to https://stackoverflow.com/questions/8724954/what-is-the-maximum-number-of-characters-for-a-host-name-in-unix and all other sources i could find, it can't exceed 255 (excluding trailing `\0`), so that would more likely indicate a bug